sdk/src/main/java/com/bugsnag/android/ThreadState.java

```diff
@@ -15,34 +15,26 @@
 class ThreadState implements JsonStream.Streamable {
     private static final String THREAD_TYPE = "android";
 
-    final Configuration config;
+    private final Configuration config;
     private final Thread[] threads;
     private final Map<Thread, StackTraceElement[]> stackTraces;
+    private final long currentThreadId;
```

Comment:
Why do we now store this?

Reply:
It's not guaranteed that `toStream` will be called on the same thread as the constructor. As we need to identify the thread where the error was originally captured, the solution I chose was to store the capturing thread ID during construction, then use it to conditionally serialise `errorReportingThread` during serialisation.

Reply:
The constructor is not guaranteed to run on the error reporting thread either. In the case of unhandled errors, the exception handler can be on a different thread altogether and this will only report the `cacheAndNotify` call and generating the `ThreadState`. The `uncaughtException()` hook provides the crashed thread which should instead be used to generate the information for the error reporting thread.

Reply:
Just checking, is this resolved now?

Reply:
@simonbowring yes, this is resolved.